plugins/redfish/fu-plugin-redfish.c

```diff
@@ -45,23 +46,139 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_redfish_client_discover_uefi_credentials (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	gsize bufsz = 0;
+	guint32 indications = 0x0;
+	g_autofree gchar *userpass_safe = NULL;
+	g_autofree guint8 *buf = NULL;
+	g_auto(GStrv) split = NULL;
+	g_autoptr(GBytes) userpass = NULL;
+
+	/* get the uint32 specifying if there are EFI variables set */
+	if (!fu_efivar_get_data (REDFISH_EFI_INFORMATION_GUID,
+				 REDFISH_EFI_INFORMATION_INDICATIONS,
+				 &buf, &bufsz, NULL, error))
+		return FALSE;
+	if (!fu_common_read_uint32_safe (buf, bufsz, 0x0,
+					 &indications, G_LITTLE_ENDIAN,
+					 error))
+		return FALSE;
+	if ((indications & REDFISH_EFI_INDICATIONS_OS_CREDENTIALS) == 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_FILE,
+				     "no indications for OS credentials");
+		return FALSE;
+	}
+
+	/* read the correct EFI var for runtime */
+	userpass = fu_efivar_get_data_bytes (REDFISH_EFI_INFORMATION_GUID,
+					     REDFISH_EFI_INFORMATION_OS_CREDENTIALS,
+					     NULL, error);
+	if (userpass == NULL)
+		return FALSE;
+
+	/* it might not be NUL terminated */
+	userpass_safe = g_strndup (g_bytes_get_data (userpass, NULL),
+				   g_bytes_get_size (userpass));
+	split = g_strsplit (userpass_safe, ":", -1);
+	if (g_strv_length (split) != 2) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INVALID_FILE,
+			     "invalid format for username:password, got '%s'",
+			     userpass_safe);
+		return FALSE;
+	}
+	fu_redfish_client_set_username (data->client, split[0]);
+	fu_redfish_client_set_password (data->client, split[1]);
+	return TRUE;
+}
+
+static gboolean
+fu_redfish_client_discover_smbios_table (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	FuContext *ctx = fu_plugin_get_context (plugin);
+	const gchar *smbios_data_fn;
+	g_autofree gchar *hostname = NULL;
+	g_autoptr(FuRedfishSmbios) redfish_smbios = fu_redfish_smbios_new ();
+	g_autoptr(GBytes) smbios_data = NULL;
+
+	/* is optional if not in self tests */
+	smbios_data_fn = g_getenv ("FWUPD_REDFISH_SMBIOS_DATA");
```

Comment:
should this maybe a fu-firmware override instead of redfish specific override?  Presumably the whole daemon should be able to run with fake smbios tables

Reply:
Yes, I need to write a `FuSmbios->write()` handler before we can do that. It's probably one for a rainy day stuck in a cabin.